src/Tree/Entity/Repository/NestedTreeRepository.php

```diff
@@ -1158,6 +1161,96 @@ public function getNodesHierarchy($node = null, $direct = false, array $options
         return $this->getNodesHierarchyQuery($node, $direct, $options, $includeNode)->getArrayResult();
     }
 
+    /**
+     * @param object      $root              Root node of the parsed tree
+     * @param object|null $node              Current node. If null, first node will be returned
+     * @param int|null    $limit             Maximum nodes to return. If null, all nodes will be returned
+     * @param string      $traversalStrategy Strategy to use to traverse tree
+     *
+     * @phpstan-assert self::TRAVERSAL_* $traversalStrategy
+     *
+     * @throws InvalidArgumentException if input is invalid
+     */
+    public function getNextNodesQueryBuilder($root, $node = null, ?int $limit = null, string $traversalStrategy = self::TRAVERSAL_PRE_ORDER): QueryBuilder
+    {
+        $meta = $this->getClassMetadata();
+        $config = $this->listener->getConfiguration($this->_em, $meta->getName());
+
+        if (self::TRAVERSAL_PRE_ORDER === $traversalStrategy) {
+            $qb = $this->childrenQueryBuilder($root, false, $config['left'], 'ASC', true);
+            if (null !== $node) {
+                $wrapped = new EntityWrapper($node, $this->_em);
+                $qb
+                    ->andWhere($qb->expr()->gt('node.'.$config['left'], ':lft'))
+                    ->setParameter('lft', $wrapped->getPropertyValue($config['left']))
+                ;
+            }
+        } elseif (self::TRAVERSAL_LEVEL_ORDER === $traversalStrategy) {
+            if (!isset($config['level'])) {
+                throw new \InvalidArgumentException('TreeLevel must be set to use level order traversal.');
```

Comment:
Why `\InvalidArgumentException`? `$config['level']` isn't provided as an argument in this method.